src/GraphQLParser/Location.cs

```diff
@@ -7,18 +7,27 @@ namespace GraphQLParser
         private static readonly Regex _lineRegex = new Regex("\r\n|[\n\r]", RegexOptions.ECMAScript);
 
         public Location(ISource source, int position)
+            : this(source.Body, position)
+        {
+           
+        }
+
+        public Location(string source, int position)
         {
             Line = 1;
             Column = position + 1;
 
-            var matches = _lineRegex.Matches(source.Body);
-            foreach (Match match in matches)
+            if (position > 0)
```

Comment:
optimization for position=0 (uninitialized location from https://github.com/graphql-dotnet/graphql-dotnet/pull/2111#discussion_r551308525)